src/permitted-handler.d.ts

```diff
@@ -0,0 +1,49 @@
+import type {
+  JsonRpcEngineEndCallback,
+  JsonRpcEngineNextCallback,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from 'json-rpc-engine';
+
+/**
+ * Makes every specified property of the specified object type mutable.
+ *
+ * @template T - The object whose readonly properties to make mutable.
+ * @template TargetKey - The property key(s) to make mutable.
+ */
+export type Mutable<
```

Comment:
I'd encourage you to add unit tests for these types as well, especially when there is something useful to assert as in this case. This is the first place we've tried writing automated tests for the types themselves! I think it has worked out pretty well so far. They would have saved us a lot of headaches, we have made _many_ type mistakes in the past few years.

Reply:
After discussing on Slack and trying several ways of testing this, we gave up